packages/build/tests/functions/tests.js

```diff
@@ -204,3 +204,39 @@ if (semver.gte(nodeVersion, '16.9.0')) {
     t.true(app2FunctionsDist.includes('worker.zip'))
   })
 }
+
+test('Functions: creates metadata file', async (t) => {
+  const fixture = await new Fixture('./fixtures/v2').withCopyRoot({ git: false })
+  const build = await fixture
+    .withFlags({
+      branch: 'my-branch',
+      cwd: fixture.repositoryRoot,
+      featureFlags: { zisi_add_metadata_file: true },
+    })
+    .runWithBuildAndIntrospect()
+
+  t.true(build.success)
+
+  const functionsDistPath = resolve(fixture.repositoryRoot, '.netlify/functions')
+  const functionsDistFiles = await readdir(functionsDistPath)
+
+  t.true(functionsDistFiles.includes('manifest.json'))
+  t.true(functionsDistFiles.includes('test.zip'))
+
+  const unzipPath = join(functionsDistPath, `.netlify-test-${Date.now()}`)
+
+  await unzipFile(join(functionsDistPath, 'test.zip'), unzipPath)
+
+  const functionFiles = await readdir(unzipPath)
+
+  t.true(functionFiles.includes('___netlify-bootstrap.mjs'))
+  t.true(functionFiles.includes('___netlify-entry-point.mjs'))
+  t.true(functionFiles.includes('___netlify-metadata.json'))
+  t.true(functionFiles.includes('test.mjs'))
+
+  const metadata = JSON.parse(await readFile(join(unzipPath, '___netlify-metadata.json'), 'utf8'))
+
+  t.is(semver.valid(metadata.bootstrap_version), metadata.bootstrap_version)
+  t.is(metadata.branch, 'my-branch')
+  t.is(metadata.version, 1)
```

Comment:
nit: should we also delete the unzipped file? or will it get cleaned up eventually?

Reply:
I don't think these directories are being cleaned up at the moment. I think that's okay since in the context of an ephemeral CI it sometimes does more harm than good, as you need to spend time deleting a bunch of files. I think it would be nice to do it optionally for a local dev setup, but I'd rather handle that more holistically as part of the `Fixture` class and not on this specific test.

Something for us to consider!